src/Cake.NuGet/Install/Extensions/NuGetPackageManagerExtensions.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.PackageManagement;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.Protocol.Core.Types;
+using NuGet.Resolver;
+
+namespace Cake.NuGet.Install.Extensions
+{
+    internal static class NuGetPackageManagerExtensions
+    {
+        public static async Task<IEnumerable<NuGetProjectAction>> GetInstallProjectActionsAsync(
+            this NuGetPackageManager packageManager,
+            NugetFolderProject nuGetProject,
+            PackageIdentity packageIdentity,
+            ResolutionContext resolutionContext,
+            INuGetProjectContext nuGetProjectContext,
+            IEnumerable<SourceRepository> sources,
+            CancellationToken token)
+        {
+            if (nuGetProject == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProject));
+            }
+
+            if (packageIdentity == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            if (resolutionContext == null)
+            {
+                throw new ArgumentNullException(nameof(resolutionContext));
+            }
+
+            if (nuGetProjectContext == null)
+            {
+                throw new ArgumentNullException(nameof(nuGetProjectContext));
+            }
+
+            if (sources == null)
+            {
+                throw new ArgumentNullException(nameof(sources));
+            }
+
+            if (!sources.Any())
+            {
+                throw new ArgumentException(nameof(sources));
+            }
+
+            if (packageIdentity.Version == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            var localSources = new HashSet<SourceRepository>(new SourceRepositoryComparer());
+            localSources.Add(packageManager.PackagesFolderSourceRepository);
+            localSources.AddRange(packageManager.GlobalPackageFolderRepositories);
```

Comment:
`GlobalPackageFolderRepositories` can't be null?

Reply:
Nope, see [this](https://github.com/NuGet/NuGet.Client/blob/release-4.3.0-rtm/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L157)